terraform/cicd-deployment/README.md

```diff
@@ -0,0 +1,442 @@
+
+## Infrastructure Components
+
+The configuration creates three main CloudFormation stacks:
+
+1. Data Exports Child Stack
+2. Data Exports Management Stack
+3. CUDOS Dashboard Stack
+
+## Prerequisites
+
+- AWS account with appropriate permissions
+- Terraform installed
+- AWS provider configured
+- Required variables defined in a `terraform.tfvars` file
+
+## Stack Details
+
+### 1. Data Exports Child Stack
+Creates a child stack for data exports aggregation with the following features:
+- Deployed in the destination account
+- Manages CUR, FOCUS, and COH data
+- Configurable resource prefix and time granularity
+- IAM capabilities enabled
+
+### 2. Data Exports Management Stack
+Creates a management stack for data exports with:
+- Dependencies on the child stack
+- Similar configuration options as the child stack
+- Management-specific resource prefixes
+- IAM capabilities enabled
+
+### 3. CUDOS Dashboard Stack
+Deploys the main CUDOS dashboard with:
+- QuickSight integration
+- Multiple dashboard deployment options
+- Cost and Usage Report (CUR) configuration
+- Optimization features
+- Technical configurations for Athena and Lake Formation
+
+## Configuration Parameters
+
+### Global Parameters
+- `destination_account_id`: Target AWS account ID
+- `source_account_ids`: List of source AWS account IDs
+- `quicksight_user`: QuickSight username
+
+### Dashboard-specific Parameters
+- CUR Configuration
+- Optimization Settings
+- Technical Settings
+- QuickSight Prerequisites
+
+## Usage
+
+1. Configure the required variables in your terraform.tfvars file
+2. Initialize Terraform:
+```bash
+terraform init
+```
+3. Review the planned changes:
+```bash
+terraform plan
+```
+4. Apply the configuration:
+```bash
+terraform apply
+```
+
+
+## Important Notes
+
+### Permissions
+- The configuration requires both `CAPABILITY_IAM` and `CAPABILITY_NAMED_IAM` permissions
+
+### Deployment Time
+- Stack creation and updates may take up to 60 minutes
+
+### Integrations
+- SNS topic integration is skipped as per checkov exception
+
+### Prerequisites
+- Ensure all required variables are properly set before deployment
+
+## Dependencies
+
+The stacks have the following hierarchical dependencies:
+```
+➤ CUDOS Dashboard
+   ↳ Data Exports Management
+   ↳ Data Exports Child
+```
+
+Stack Creation Order:
+1. Data Exports Child
+2. Data Exports Management
+3. CUDOS Dashboard
+
+
+
+- Data Exports Management depends on Data Exports Child
+- CUDOS Dashboard depends on both Data Exports Management and Child stacks
+
+## Alternative Manual Deployment
+
+If you don't have a pipeline configured, you can split the deployment into three separate stages. This approach requires manually deploying the components in sequence:
+
+### Directory Structure for Manual Deployment
+Split the existing code into three separate folders:
+
+```
+project-root/
+├── 1-data-exports-child/
```

Comment:
child is not the therminology we use. Typically it is Source and Destination or Management Account vs Data Collection Account